test/versioning/npm.spec.js

```diff
@@ -123,4 +123,88 @@ describe('semver.getNewValue()', () => {
       semver.getNewValue('~0.6.1', 'replace', '0.6.8', '0.7.0-rc.2')
     ).toEqual('~0.7.0-rc');
   });
+  it('handles cap range strategy', () => {
+    [
+      ['=1.0.0', '1.0.0', '1.2.3', '=1.2.3'],
+      ['=1.2', '1.0.0', '1.2.3', '=1.2 <=1.2.3'], // '>=1.2.0 <1.3.0'
+      ['=1', '1.0.0', '1.2.3', '=1 <=1.2.3'], // '>=1.0.0 <2.0.0'
+
+      ['1.0.0 || 1.1.0', '1.0.0', '1.2.3', '1.0.0 || 1.2.3'],
+      ['1.0.0', '1.0.0', '1.2.3', '1.2.3'],
+      ['1.2', '1.0.0', '1.2.3', '1.2 <=1.2.3'], // '>=1.2.0 <1.3.0'
+      ['1', '1.0.0', '1.2.3', '1 <=1.2.3'], // '>=1.0.0 <2.0.0'
+
+      ['^1', '1.0.0', '1.3.0', '^1 <=1.3.0'], // '>=1.0.0 <2.0.0'
+      ['^1.2', '1.2.0', '1.3.0', '^1.2 <=1.3.0'], // '>=1.2.0 <2.0.0'
+      ['^1.2.3', '1.2.3', '1.3.0', '^1.2.3 <=1.3.0'], // '>=1.2.3 <2.0.0'
+
+      ['~>1', '1.0.0', '1.2.0', '~>1 <=1.2.0'], // '>=1.0.0 <2.0.0'
+      ['~>1.2', '1.2.0', '1.3.0', '>=1.2.0 <=1.3.0'], // '>=1.2.0 <1.3.0'
+      ['~>1.2.3', '1.2.3', '1.3.0', '>=1.2.3 <=1.3.0'], // '>=1.2.3 <1.3.0'
+
+      ['~1', '1.0.0', '1.2.0', '~1 <=1.2.0'], // '>=1.0.0 <2.0.0'
+      ['~1.2', '1.2.0', '1.3.0', '>=1.2.0 <=1.3.0'], // '>=1.2.0 <1.3.0'
+      ['~1.2.3', '1.2.3', '1.3.0', '>=1.2.3 <=1.3.0'], // '>=1.2.3 <1.3.0'
+
+      ['1.*', '1.0.0', '1.2.3', '1.* <=1.2.3'], // '>=1.0.0 <2.0.0'
+      ['1.*', '1.0.0', '2.3.4', '>=1.0.0 <=2.3.4'], // '>=1.0.0 <2.0.0'
+      ['1.2.*', '1.2.0', '1.2.3', '1.2.* <=1.2.3'], // '>=1.0.0 <1.3.0'
+      ['1.2.*', '1.0.0', '2.3.4', '>=1.2.0 <=2.3.4'], // '>=1.0.0 <1.3.0'
+
+      ['1.x', '1.0.0', '1.2.3', '1.x <=1.2.3'], // '>=1.0.0 <2.0.0'
+      ['1.x', '1.0.0', '2.3.4', '>=1.0.0 <=2.3.4'], // '>=1.0.0 <2.0.0'
+      ['1.2.x', '1.2.0', '1.2.3', '1.2.x <=1.2.3'], // '>=1.0.0 <1.3.0'
+      ['1.2.x', '1.0.0', '2.3.4', '>=1.2.0 <=2.3.4'], // '>=1.0.0 <1.3.0'
+
+      ['>=1.0.0 <=1.2.3', '1.0.0', '1.2.4', '>=1.0.0 <=1.2.4'],
+      [
+        '>=1.0.0 <=1.2.3 || >=1.3.0',
+        '1.0.0',
+        '1.4.0',
+        '>=1.0.0 <=1.2.3 || >=1.3.0 <=1.4.0',
+      ],
+      [
+        '>=1.0.0 <=1.2.3 || 1.3.0',
+        '1.0.0',
+        '1.4.0',
+        '>=1.0.0 <=1.2.3 || 1.4.0',
+      ],
+
+      ['>=1.0.0 <=1.2.3', '1.0.0', '1.5.0', '>=1.0.0 <=1.5.0'],
+      ['^1.0', '1.0.0', '1.0.7', '^1.0 <=1.0.7'],
+      ['^1', '1.0.0', '1.0.7-prerelease.1', '^1 <=1.0.7-prerelease.1'],
```

Comment:
I don't think that `1.0.7-prerelease.1` falls within the range `^1` so this may not be a valid use case.